managed/services/dump/dump.go

```diff
@@ -113,6 +113,28 @@ func (s *Service) StartDump(params *Params) (string, error) {
 	s.cancel = cancel
 	s.rw.Unlock()
 
+	// include both service names and node names in the instances flag
+	// that way, pmm-dump is able to export both OS metrics and QAN metrics.
+	instances := make([]string, 0, len(params.ServiceNames))
+	nodeNames := make(map[string]struct{})
+	for _, serviceName := range params.ServiceNames {
+		service, err := models.FindServiceByName(s.db.Querier, serviceName)
+		if err != nil {
+			s.running.Store(false)
+			return "", errors.Wrapf(err, "failed to find service %s", serviceName)
+		}
+		node, err := models.FindNodeByID(s.db.Querier, service.NodeID)
+		if err != nil {
+			s.running.Store(false)
+			return "", errors.Wrapf(err, "failed to find node for service %s", serviceName)
+		}
+		instances = append(instances, serviceName)
+		nodeNames[node.NodeName] = struct{}{}
+	}
+	for nodeName := range nodeNames {
```

Comment:
IMO we don't need this extra `for` loop, it can be done in the loop above.

Reply:
Not really, it ensures that there are no duplicates. But I'm not sure how critical is that

Reply:
Ok, not critical )